tempesta_fw/http_msg.c

```diff
@@ -845,15 +846,16 @@ EXPORT_SYMBOL(tfw_http_msg_free);
  * That allows for a shorter (limited) initialization.
  */
 TfwHttpMsg *
-tfw_http_msg_alloc_err_resp(void)
+tfw_http_msg_alloc_err_resp(TfwHttpReq *req)
 {
-	TfwHttpMsg *hm;
+	TfwHttpResp *resp;
 
-	if (!(hm = (TfwHttpMsg *)tfw_pool_new(TfwHttpResp, TFW_POOL_ZERO)))
+	if (!(resp = (TfwHttpResp *)tfw_pool_new(TfwHttpResp, TFW_POOL_ZERO)))
 		return NULL;
-	ss_skb_queue_head_init(&hm->msg.skb_list);
+	tfw_http_msg_pair((TfwHttpMsg *)req, (TfwHttpMsg *)resp);
```

Comment:
Not sure that it is worth to combine response allocation and request pairing: in https://github.com/tempesta-tech/tempesta/pull/877 interface `tfw_http_msg_alloc_err_resp()` is replaced by light version of general message allocation interface. So, as a result - pairing functionality will still have to be taken out separately (or to be added to general `__tfw_http_msg_alloc` interface, which is undesirable).

Reply:
The reason to combine allocation and pairing is to force pairing of request and responses. In the patch response without paired request is a bug. To avoid creating responses without paired request a new interface is introduced.